src/ILICheck.Web/ClientApp/src/home.js

```diff
@@ -61,75 +51,46 @@ export const Home = (props) => {
 
-  useEffect(() => {
-    if (validationResult !== "none") {
-      let className;
-      let text;
-      let downloadAvailable = false;
-      setTestRunning(false);
-
-      if (validationResult === "ok") {
-        downloadAvailable = true;
-        className = "valid";
-        text = "Keine Fehler!";
-        updateLog("Die Daten sind modellkonform!");
-      }
-
-      if (validationResult === "error") {
-        downloadAvailable = true;
-        className = "errors";
-        text = "Fehler!";
-        updateLog("Die Daten sind nicht modellkonform! Für Fehlermeldungen siehe XTF-Log-Datei.");
-      }
-
-      if (validationResult === "aborted") {
-        className = "errors";
-        text = "Fehler!";
-        updateLog("Die Validierung wurde abgebrochen.");
-      }
-
-      setFileCheckStatus({
-        text: text,
-        class: className,
-        testRunTime: new Date(),
-        fileName: fileToCheck ? fileToCheck.name : "",
-        fileDownloadAvailable: downloadAvailable,
-      });
-
-      setValidationResult("none");
-    }
-  }, [validationResult, fileToCheck, setValidationResult, updateLog, clientSettings]);
-
   const checkFile = (e) => {
     e.stopPropagation();
     resetLog();
     setTestRunning(true);
-    setFileCheckStatus({ text: "", class: "", testRunTime: null, fileName: "", fileDownloadAvailable: false });
     setUploadLogsInterval(setIntervalImmediately(logUploadLogMessages, 2000));
     uploadFile(fileToCheck);
   };
 
-  const uploadFile = (file) => {
+  async function uploadFile(file) {
     const formData = new FormData();
-    formData.append(file.name, file);
-
-    const controller = new AbortController();
-    const signal = controller.signal;
-    fetch(`api/v1/upload?connectionId=${connection.connectionId}&fileName=${file.name}`, {
+    formData.append("file", file, file.name);
+    const response = await fetch(`api/v1/upload`, {
       method: "POST",
-      signal: signal,
       body: formData,
-    })
-      .then((res) => {
-        if (res.status === 200) {
-          console.log("Datei erfolgreich hochgeladen.");
-        } else {
-          updateLog("Fehler beim Hochladen der Datei.");
-          console.log("Fehler beim Hochladen der Datei.");
+    });
+    if (response.ok) {
+      const data = await response.json();
+      var interval = setInterval(async () => {
+        const status = await fetch(data.statusUrl, {
+          method: "GET",
+        });
+        const statusData = await status.json();
+        updateLog(statusData.statusMessage);
+        if (
+          statusData.status === "completed" ||
+          statusData.status === "completedWithErrors" ||
+          statusData.status === "failed"
+        ) {
+          clearInterval(interval);
+          setTestRunning(false);
+          setStatusData(statusData);
         }
-      })
-      .catch((err) => console.error(err));
-  };
+      }, 1000);
```

Comment:
PP: Wenn möglich direkt nach dem Upload die Status API aufrufen, damit der Status 'wird validiert...' nicht verpasst wird. Danach könnte das Polling allenfalls auf 2 sec erhöt werden. Für mich passt aber die aktuelle Lösung auch 👍🏼 Es ist unwahrscheinlich, dass der gesamte Prozess serverseitig in unter einer Sekunde abgeschlossen weden kann. Das Risiko, dass die entsprechende Meldung möglicherweise fehlt, ist auf jeden Fall vertretbar.